manifests/config/server/service.pp

```diff
@@ -37,19 +41,22 @@
   }
 
   if $service_ensure =~ /^(absent|false)$/ {
-    $changes = "rm Server/Service[#attribute/name='${name}']"
+    $changes = "rm Server/Service[#attribute/name='${service_name}']"
   } else {
-    if $class_name_ensure =~ /^(absent|false)$/ {
-      $_class_name = "rm Server/Service[#attribute/name='${name}']/#attribute/className"
-    } elsif $class_name {
-      $_class_name = "set Server/Service[#attribute/name='${name}']/#attribute/className ${class_name}"
+    if $class_name_ensure =~ /^(present|true)$/ {
+      if empty($class_name) {
+        fail('$class_name must be specified when $class_name_ensure is set to true or present')
+      }
+      $_class_name = "set Server/Service[#attribute/name='${service_name}']/#attribute/className ${class_name}"
+    } else {
+      $_class_name = "rm Server/Service[#attribute/name='${service_name}']/#attribute/className"
```

Comment:
Oh I see, you're changing the default value above, so I guess the behavior would change only when someone was specifying `class_name` and not `class_name_ensure` in that before it would do nothing and after would raise a helpful error message.